Code/Framework/AzCore/AzCore/Instance/InstancePool.h

```diff
@@ -108,8 +117,16 @@ namespace AZ
                 else
                 {
                     m_nameToInstancePool.erase(insertResult.first);
-                    m_nameToInstancePool.insert({ name, instancePool });
-                    return AZ::Success(instancePool);
+                    insertResult = m_nameToInstancePool.insert({ name, instancePool });
+
+                    if (insertResult.second)
+                    {
+                        return AZ::Success(instancePool);
+                    }
+                    else
+                    {
+                        return AZ::Failure(AZStd::string::format("Could not add InstancePool with name %.s", name.GetCStr()));
```

Comment:
I guess it was trying to put the dot at the end of the sentence? `%.s` > `%s.`

Reply:
good catch! Fixed.